src/prommis/nanofiltration/costing/diafiltration_cost_model.py

```diff
@@ -440,3 +441,98 @@ def variable_operating_cost_constraint(blk):
                 to_units=blk.costing_package.base_currency
                 / blk.costing_package.base_period,
             )
+
+    def cost_precipitator(
+        blk,
+        precip_volume,
+    ):
+        """
+        Costing method for precipitator unit. Assumes these are horizontal vessels
+        made from 1.25 in thick carbon steel, includes platforms and ladders, and each
+        instance is one unit (default args).
+
+        References:
+            residence time:
+                https://www.sciencedirect.com/science/article/pii/S0304386X19309806
+                https://onlinelibrary.wiley.com/doi/full/10.1002/ceat.201700667
+                https://pubs.acs.org/doi/full/10.1021/acs.iecr.1c04876
+                https://www.sciencedirect.com/science/article/pii/S0304386X01002134
+            vessel dimensions:
+                https://www.accessengineeringlibrary.com/content/book/9781260455410/back-matter/appendix1?implicit-login=true
+
+        Args:
+            precip_volume: volume of the precipitator as calculated by the unit model (m3)
+        """
+
+        # calculate the volume needed
+        blk.volume_capacity = Var(
+            initialize=120,
+            domain=NonNegativeReals,
+            doc="Volume requirement of precipitator vessel",
+            units=units.m**3,
+        )
+
+        # account for a 20% headspace
+        @blk.Constraint()
+        def volume_capacity_equation(blk):
+            return blk.volume_capacity == units.convert(
+                (1.2 * precip_volume), to_units=units.m**3
+            )
+
+        # include a length and diameter constraint
+        # TODO: L and D should get bounded but gives init errors
+        blk.precipitator_diameter = Var(
+            initialize=6,
+            domain=NonNegativeReals,
+            doc="Diameter of the precipitator vessel",
+            units=units.ft,
+        )
+        blk.precipitator_diameter.fix()
+        blk.precipitator_length = Var(
+            initialize=8,
+            domain=NonNegativeReals,
+            doc="Length of the precipitator vessel",
+            units=units.ft,
+        )
+
+        @blk.Constraint()
+        def diameter_length_ratio_equation(blk):
+            return units.convert(blk.precipitator_length, to_units=units.inch) == (
+                units.convert(
+                    (
+                        blk.volume_capacity
+                        - units.convert(
+                            (
+                                2
+                                * 0.954
```

Comment:
From the vessel dimensions link in the docstring, it appears that the formula gives volume in gallons:

```suggestion
                                * 0.954 * pyunits.gal / pyunits.in**3
```